### PR TITLE
Dhis2 intergration

### PR DESCRIPTION
**Pull request by solemabrothers**
@ssmusoke here is the PR:
To reproduce the error, follow the steps below
1. Generate a JSON Report
2. Click the Button To DHIS2
3. On the Preview Window, click Send to DHIS2 Button,
Read the error log

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    }
+
+
+
+    private SimpleObject sendData(byte[] data) throws IOException {
+        Map map = new HashMap();
+        SimpleObject simpleObject=new SimpleObject();
+        int responseCode = 0;
+        try {
+            URL url = new URL("https://ugisl.mets.or.ug:5000/ehmis");
```
Comment (ssmusoke) [inline omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java]: @solemabrothers Do not harcode the url to which the connection is being made

File: omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+            String encoding = Base64.getEncoder().encodeToString(("mets.mkaye:METS4321!").getBytes("UTF-8"));
```
Comment (ssmusoke) [inline omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java]: @solemabrothers Do not hardcode the username and password to the HIM

File: omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.module.reporting.report.ReportRequest;
+import org.openmrs.module.reporting.report.renderer.RenderingMode;
+import org.openmrs.module.reporting.report.service.ReportService;
+import org.openmrs.module.reporting.web.renderers.WebReportRenderer;
+import org.openmrs.ui.framework.SimpleObject;
+import org.openmrs.ui.framework.annotation.SpringBean;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.net.ssl.*;
```
Comment (ssmusoke) [inline omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java]: @solemabrothers Remove all wildcard imports

File: omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java (hunk 2 above)
Comment (solemabrothers) [reply omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java]: In your opinion what’s the best way to pass those credentials to the connection object.

**Comment by solemabrothers** [comment]
@ssmusoke this PR depends on the Sync PR here: https://github.com/METS-Programme/openmrs-module-ugandaemr-sync/pull/25

File: omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java (hunk 1 above)
Comment (solemabrothers) [reply omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java]: Removed

File: omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java (hunk 2 above)
Comment (solemabrothers) [reply omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java]: Removed

File: omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java (hunk 3 above)
Comment (solemabrothers) [reply omod/src/main/java/org/openmrs/module/ugandaemrreports/fragment/controller/SendReportRequestFragmentController.java]: Removed

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 4]
```diff
@@ -26,7 +26,7 @@
 import org.openmrs.module.ugandaemrreports.reporting.calculation.smc.SmcReturnFollowUpCalculation;
 import org.openmrs.module.ugandaemrreports.reporting.cohort.definition.CalculationCohortDefinition;
 import org.openmrs.module.ugandaemrreports.reporting.metadata.Dictionary;
-import org.openmrs.module.ugandaemrreports.reporting.metadata.Metadata;
+import org.openmrs.module.ugandaemrreports.reporting.metadata.*;
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @solemabrothers Remove wildcard import

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 5]
```diff
@@ -555,8 +559,8 @@ public CohortDefinition testedHivNegative() {
      * Ever Tested for HIV Before
      * @return CohortDefinition
      */
-    public CohortDefinition everTestedForHivBefore() {
-        return definitionLibrary.hasObs(Dictionary.getConcept(Metadata.Concept.HIV_TEST), Dictionary.getConcept(Metadata.Concept.YES_CIEL));
+    public CohortDefinition individualsTestingFortheFirstTime() {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @solemabrothers Why delete the old definition which is used in previous reports? They are now broken

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 6]
```diff
@@ -571,10 +575,12 @@ public CohortDefinition testedInLast12Months() {
      * Tested more than twice in the last 12 Months
      * @return CohortDefinition
      */    
-	public CohortDefinition testedMoreThanTwiceInLast12Months() {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @solemabrothers Why delete the old definition which is used in previous reports? They are now broken. Just create a new definition and leave the old one untouched

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 7]
```diff
@@ -614,16 +650,16 @@ public CohortDefinition individualsWhoReceivedHIVTestResultsInLast12Months() {
      * Individuals Tested for the first time
      * @return CohortDefinition
      */
-    public CohortDefinition individualsTestedFirstTime() {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @solemabrothers Why delete the old definition which is used in previous reports? They are now broken. Just create a new definition and leave the old one untouched

File: api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java [hunk 8]
```diff
@@ -668,16 +703,16 @@ public CohortDefinition tbSuspect() {
 	 * Individuals tested more than twice in the last 12 months
 	 * @return CohortDefinition
 	 */			
-	public CohortDefinition individualsTestedMoreThanTwiceInLast12Months() {
```
Comment (ssmusoke) [inline api/src/main/java/org/openmrs/module/ugandaemrreports/library/Moh105CohortLibrary.java]: @solemabrothers Why delete the old definition which is used in previous reports? They are now broken. Just create a new definition and leave the old one untouched

**Comment by ssmusoke** [comment]
@solemabrothers For this PR do not change definitions for existing reports, create new ones - as we may need those reports for the old data

**Pull request closed without merging**